astropy/modeling/core.py

```diff
@@ -227,13 +227,7 @@ def rename(cls, name):
 
         new_cls = type(name, (cls,), {})
         new_cls.__module__ = modname
-
-        if hasattr(cls, '__qualname__'):
-            if new_cls.__module__ == '__main__':
-                # __main__ is not added to a class's qualified name
-                new_cls.__qualname__ = name
-            else:
-                new_cls.__qualname__ = '{0}.{1}'.format(modname, name)
+        new_cls.__qualname__ = name
```

Comment:
This was a bug - ``__qualname__`` should not include the module name (see [PEP 3155](https://www.python.org/dev/peps/pep-3155/))